src/shared/sleep-config.c

```diff
@@ -127,6 +133,50 @@ int parse_sleep_config(SleepConfig **ret_sleep_config) {
         return 0;
 }
 
+static int acquire_bus(sd_bus **bus) {
+        int r;
+
+        assert(bus);
+
+        if (*bus)
+                return 0;
+
+        r = bus_connect_transport(BUS_TRANSPORT_LOCAL, NULL, false, bus);
+        if (r < 0)
+                return bus_log_connect_error(r, BUS_TRANSPORT_LOCAL);
+
+        return 0;
+}
+
+/* Freeze or thaw user.slice */
+int freeze_thaw_user_slice(const char **method) {
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *m = NULL;
+        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+        sd_bus *bus;
+        int r;
+
+        if (*method == NULL)
+                return -1;
```

Comment:
this is not an error, return 0 - also check `method`, not `*method`

Reply:
if i return 0, method will not be set to NULL to skip cleanup

Reply:
how so?

Reply:
because it checks fir r<0, and not r <= 0.

Reply:
It doesn't matter, because it doesn't pass null at that point

Reply:
ok